rosidl_typesupport_opensplice_c/resource/msg__type_support_c.cpp.template

```diff
@@ -69,30 +69,40 @@ ROSIDL_GET_TYPE_SUPPORT_FUNCTION(@(pkg), @(subfolder), @(msg))();
 have_not_included_primitive_arrays = True
 have_not_included_string = True
 }@
-@[for field in spec.fields]@
-@[if field.type.is_primitive_type()]@
-@[if field.type.is_array and have_not_included_primitive_arrays]@
+{
+from collections import OrderedDict
+field_types = {}
+for field in spec.fields:
+    if not field.type.is_primitive_type():
+        full_field = field.type.pkg_name + '__msg__' + field.type.type
```

Comment:
I meant `BaseType.type`. Anyway, it's weird, but I didn't get unique elements in the set, so I thought `BaseType` didn't implement `__hash__`, but I see it does. I've updated the code to use `OrderedSet`